node/lib/util/push.js

```diff
@@ -84,23 +84,75 @@ exports.push = co.wrap(function *(repo, remoteName, source, target, force) {
     let remoteUrl = yield GitUtil.getUrlFromRemoteName(repo, remoteName);
 
     // First, push the submodules.
+    const annotatedCommit = yield GitUtil.resolveCommitish(repo, source);
+    const sha = annotatedCommit.id();
+    const commit = yield repo.getCommit(sha);
+
+    const submoduleSet = new Set(yield SubmoduleUtil.listOpenSubmodules(repo));
```

Comment:
Is there a way you could construct this `Set` from the contents of two arrays without using a loop?

Reply:
Not necessarily; it seems as though there should be a better way but I can't think of one either.